exporter/exporterhelper/internal/queue_capacity.go

```diff
@@ -0,0 +1,73 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package internal // import "go.opentelemetry.io/collector/exporter/exporterhelper/internal"
+
+import (
+	"sync/atomic"
+)
+
+type itemsCounter interface {
+	ItemsCount() int
+}
+
+// sizeOfFunc is a function that returns the size of the given element.
+type sizeOfFunc[T any] func(T) uint64
+
+type QueueCapacityLimiter[T any] struct {
```

Comment:
Instead of exposing this, can we just pass a capacity and a sizer to the queue constructor?

Reply:
done